packages/framework/tests/Unit/Model/Order/Processing/OrderProcessorMiddleware/SetDeliveryAddressByDeliveryAddressUuidMiddlewareTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Unit\Model\Order\Processing\OrderProcessorMiddleware;
+
+use Shopsys\FrameworkBundle\Model\Customer\Customer;
+use Shopsys\FrameworkBundle\Model\Customer\DeliveryAddress;
+use Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressFacade;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser;
+use Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessorMiddleware\SetDeliveryAddressByDeliveryAddressUuidMiddleware;
+use Tests\FrameworkBundle\Test\MiddlewareTestCase;
+
+class SetDeliveryAddressByDeliveryAddressUuidMiddlewareTest extends MiddlewareTestCase
+{
+    public function testTransportIsAdded(): void
+    {
+        $orderProcessingData = $this->createOrderProcessingData();
+
+        $deliveryAddressUuid = '30a07736-85aa-4407-8a85-4e722ed3e035';
+        $customerUser = $this->createCustomerUser();
+
+        $orderProcessingData->orderInput->addAdditionalData(SetDeliveryAddressByDeliveryAddressUuidMiddleware::DELIVERY_ADDRESS_UUID, $deliveryAddressUuid);
+        $orderProcessingData->orderInput->setCustomerUser($customerUser);
+
+        $expectedFirstName = 'firstName';
+        $expectedLastName = 'lastName';
+        $expectedCompanyName = 'companyName';
+        $expectedTelephone = '123456789';
+        $expectedStreet = 'street';
+        $expectedCity = 'city';
+        $expectedPostCode = '12312';
+        $expectedCountry = 'CZ';
+
+        $deliveryAddress = $this->createMock(DeliveryAddress::class);
+        $deliveryAddress->method('getFirstName')->willReturn($expectedFirstName);
+        $deliveryAddress->method('getLastName')->willReturn($expectedLastName);
+        $deliveryAddress->method('getCompanyName')->willReturn($expectedCompanyName);
+        $deliveryAddress->method('getTelephone')->willReturn($expectedTelephone);
+        $deliveryAddress->method('getStreet')->willReturn($expectedStreet);
+        $deliveryAddress->method('getCity')->willReturn($expectedCity);
+        $deliveryAddress->method('getPostcode')->willReturn($expectedPostCode);
+        $deliveryAddress->method('getCountry')->willReturn($expectedCountry);
```

Comment:
would not it be easier to create and actual instance using `DeliveryAddressData` instance?

Reply:
true, changed :)